mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryMailboxMapper.java

```diff
@@ -134,12 +135,15 @@ public Flux<Mailbox> list() {
 
     @Override
     public Mono<ACLDiff> updateACL(Mailbox mailbox, MailboxACL.ACLCommand mailboxACLCommand) {
-        return Mono.fromCallable(() -> {
-            MailboxACL oldACL = mailbox.getACL();
-            MailboxACL newACL = mailbox.getACL().apply(mailboxACLCommand);
-            mailboxesByPath.get(mailbox.generateAssociatedPath()).setACL(newACL);
-            return ACLDiff.computeDiff(oldACL, newACL);
-        });
+        return Mono.just(mailbox.getACL())
+            .flatMap(oldACL -> Mono.fromCallable(() -> mailboxesByPath.compute(mailbox.generateAssociatedPath(), (path, existingMailbox) -> {
+                    if (existingMailbox == null) {
+                        throw new IllegalArgumentException("Mailbox not found for path: " + path);
+                    }
+                    Throwing.runnable(() -> existingMailbox.setACL(existingMailbox.getACL().apply(mailboxACLCommand))).run();
```

Comment:
I prefer a try-catch block rather than this half functional code.

Reply:
this function throw check exception (`UnsupportedRightException`)
If use try-catch in lamda, it should still be wrapped in a RuntimeException.